_posts/java/2023-01-18-RSPEC-3749.md

```diff
@@ -0,0 +1,50 @@
+---
+title: 스프링 컴포넌트의 변수들은 주입되어야합니다.
+tags: [java, spring, owasp, Vulnerability]
+
+스프링의 `@Component`, `@Controller`, `@Service` 그리고 `@Repository` 클래스들은 기본적으로 애플리케이션에서 하나만 인스턴스화 되는 싱글톤 객체입니다.
+일반적으로 이러한 클래스들은 `logger`와 같은 적은 수의 `static` 멤버 변수들을 가질 수 있지만, 모든 `non-static` 멤버 변수들은 스프링에 의해 관리되어야 합니다.
+즉, 이러한 변수들은 `@Resource`, `@Inject` 와 같은 종류의 어노테이션이 달려있어야 합니다.
+
+이러한 클래스들에 주입되지 않은 멤버 변수를 가지는 것은 상태를 변경하려는 시도를 나타낼 수 있습니다.
```

Comment:
뒤쪽 문장이 누락된 것 같아요. 확인 부탁드립니다.

> Because they are singletons, such an attempt is almost guaranteed to eventually expose data from User1’s session to User2.